modin/experimental/backends/omnisci/query_compiler.py

```diff
@@ -86,6 +86,8 @@ class DFAlgQueryCompiler(BaseQueryCompiler):
     def __init__(self, frame, shape_hint=None):
         assert frame is not None
         self._modin_frame = frame
+        if shape_hint is None and len(self._modin_frame.columns) == 1:
```

Comment:
Should we add the following check?
```python
if shape_hint is None and len(self._modin_frame.index) == 1:
    shape_hint = "row"
```

Reply:
No, the meaning of the `shape_hint` is to avoid triggering execution to get information about the frame shape. Accessing the frame's index will trigger the whole delayed computation tree.